### DISCUSSION TITLE
Cannot upgrade firmware for  Shelly Plus 1PM from V12.5.0 to V13.0.0

### DISCUSSION THREAD

**Discussion by wjohn007** [arendst/Tasmota#19162] [Q&A] [answered]
I tried to install Tasmota on  my  Shelly Plus 1PM. 

The partitions after changing from MgOS32 to Tasmota.

![grafik](https://github.com/arendst/Tasmota/assets/115527085/14a24634-c4dc-4d07-ae2f-9df2102c7179)

The partition wizard shows following
![grafik](https://github.com/arendst/Tasmota/assets/115527085/5f0b1c91-0db8-41db-959d-81cfb7dabd6a)

After  "Start migration" I still see version V12.5.0.

What do I have to change?

Kind regards
John

**Comment by Jason2866** [comment, marked as answer]
This is a bit tricky, you need to switch the active partition where Tasmota is started from.
To do this do an upgrade with the current version. So copy `http://ota.tasmota.com/tasmota32/release-12.5.0/tasmota32solo1.bin` as OTA URL and hit upgrade. After that the red `ERROR` should be gone.
After that the conversion process should work.

II is explained in the documentation https://github.com/tasmota/mgos32-to-tasmota32#convert-to-tasmota-safeboot

**Reply by wjohn007** [reply]
Thanks a lot. Everything works as expected now.

John
